src/keyspace.rs

```diff
@@ -372,7 +372,13 @@ impl Keyspace {
         let mut partitions = self.partitions.write().expect("lock is poisoned");
 
         Ok(if let Some(partition) = partitions.get(name) {
-            partition.clone()
+            let handle = partition.clone();
+            /// throw error if [PARTITION_DELETED_MARKER] exists with a existing partition
+            if handle.path().join(PARTITION_DELETED_MARKER).exists() {
+                log::error!("Failed to open partition, partition is deleted.");
+                return Err(Error::PartitionDeleted);
+            }
```

Comment:
This should be in the `else` case.

When the partition is deleted, it is not part of `partitions` anymore. I guess there could be a race condition, where `Keyspace::delete_partition` is not finished yet, but that could be fixed by wrapping the entire function in the write lock.

Anyway, if you create a new test file, do something like this:

```rs
let keyspace = ...;
let partition = keyspace.open_partition("default", Default::default())?;
keyspace.delete_partition(partition.clone())?;

// This should throw the PartitionDeleted error variant...
assert!(keyspace.open_partition("default", Default::default).is_err());

// Dropping the partition should delete folder, so we should be able to recreate it again now
drop(partition);
assert!(keyspace.open_partition("default", Default::default).is_ok());
```

Reply:
My bad. I thought so, it should be in the else case must have interpreted the comments wrong. I will refactor

Reply:
I have moved this sanity check while createing a partition and added a test check to verify the behaviour. Thanks for your help